src/mqt/bench/benchmarks/shor.py

```diff
@@ -31,7 +31,8 @@
 
 import numpy as np
 from qiskit.circuit import Instruction, ParameterVector, QuantumCircuit, QuantumRegister
-from qiskit.circuit.library import QFT
+from qiskit.circuit.library import QFTGate
+from qiskit.synthesis import synth_qft_full
```

Comment:
Isn't it slightly inconsistent to use both of these methods here instead of just one of them? Also holds for other places where either one of these is used.

Reply:
Ah. In that case I think it might actually make sense to leave it as is and use `QFTGate` wherever that is enough and only use `synth_qft_full` when needed.
There are some high-level synthesis plugins that might take the QFT Gate and synthesize it differently. I remember reading something about that.